x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/job_creator.ts

```diff
@@ -293,7 +293,11 @@ export class JobCreator {
   public set useDedicatedIndex(enable: boolean) {
     this._useDedicatedIndex = enable;
     if (enable) {
-      this._job_config.results_index_name = this._job_config.job_id;
+      if (this._job_config.results_index_name === undefined) {
```

Comment:
If I edit this in the JSON on the final summary step of the wizard, but then go back two steps to the job details page, _don't_ make an edit to the job ID, and then go forward again, the custom name has been reset to the default based on the job ID. An unlikely scenario, but is there anything that can be done to prevent it being reset if the user goes back to a previous step?

Reply:
To fix this issue, we'd need to change the way we're setting the dedicated index. We'd need to add an ability to keep track of whether the user has changed the `results_index_name` value while editing the JSON.

As a work around, I've added the name of the index to the component's description in the UI. This way the user will be able to spot if their overridden value has been reset while navigating backwards in the wizard.
![image](https://user-images.githubusercontent.com/22172091/214340926-560d5f2e-1a4d-4c30-8e73-e3fd920a9766.png)